src/main/java/uk/gov/justice/hmpps/prison/repository/sql/InmateRepositorySql.kt

```diff
@@ -185,11 +185,14 @@ enum class InmateRepositorySql(val sql: String) {
         IST.BAND_CODE as BAND_CODE,
         OIS.IMPRISONMENT_STATUS as IMPRISONMENT_STATUS
         FROM OFFENDER_BOOKINGS OB
-        INNER JOIN OFFENDERS O ON OB.OFFENDER_ID = O.OFFENDER_ID
-                LEFT JOIN AGENCY_INTERNAL_LOCATIONS AIL ON OB.LIVING_UNIT_ID = AIL.INTERNAL_LOCATION_ID
-                LEFT JOIN OFFENDER_IMPRISON_STATUSES OIS ON OIS.OFFENDER_BOOK_ID = OB.OFFENDER_BOOK_ID AND OIS.LATEST_STATUS = 'Y'
-        LEFT JOIN IMPRISONMENT_STATUSES IST ON IST.IMPRISONMENT_STATUS = OIS.IMPRISONMENT_STATUS
-                WHERE OB.ACTIVE_FLAG = 'Y' AND OB.BOOKING_SEQ = 1
+            INNER JOIN OFFENDERS O ON OB.OFFENDER_ID = O.OFFENDER_ID
+            LEFT JOIN AGENCY_INTERNAL_LOCATIONS AIL ON OB.LIVING_UNIT_ID = AIL.INTERNAL_LOCATION_ID
+            LEFT JOIN (
+                SELECT OIST.* FROM OFFENDER_IMPRISON_STATUSES OIST WHERE OIST.IMPRISON_STATUS_SEQ IN 
+                (SELECT MAX(IMPRISON_STATUS_SEQ) FROM OFFENDER_IMPRISON_STATUSES WHERE LATEST_STATUS = 'Y' AND OFFENDER_BOOK_ID = OIST.OFFENDER_BOOK_ID)
```

Comment:
Is this at risk of reaching the maximum values passed into an IN clause? I think it's a 1000.